File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/note/presenter/NoteContract.java
package com.github.abhijitpparate.keeper.screen.note.presenter;


import android.support.annotation.ColorRes;
import android.support.annotation.StringRes;

import com.github.abhijitpparate.checklistview.CheckListItem;
import com.github.abhijitpparate.keeper.data.database.Note;
import com.google.android.gms.location.places.Place;

import java.util.List;

public interface NoteContract {

    public enum NoteColor {
        DEFAULT, WHITE, PURPLE, RED, GREEN, YELLOW, BLUE, ORANGE
    }

    interface View {

        String getNoteId();
        String getNoteTitle();
        String getNoteBody();
        String getCheckList();
        String getNoteColor();
        Note.Place getNotePlace();

        void setNote(Note notes);
        void setNoteTitle(String title);
        void setNoteBody(String body);
        void setNoteChecklist(List<CheckListItem> checklist);
        void setNoteColor(@ColorRes int colorInt, String colorString);
        void setNotePlace(Note.Place place);

        void showOptionsPanel();
        void hideOptionsPanel();
        void showPlacePicker();

        void showHomeScreen();

        void loadNoteIfAvailable();

        void setPresenter(NoteContract.Presenter presenter);
        void showProgressbar(boolean bool);
        void makeToast(@StringRes int strId);
        void makeToast(String message);

        void switchToChecklist();
        void switchToText();
    }

    interface Presenter {
        void onSaveClick();

        void loadNote(String noteId);

        void subscribe();
        void unsubscribe();

        void onChecklistClick(boolean isChecked);
        void onDrawingClick();
        void onLocationClick();
        void onAudioClick();
        void onVideoClick();
        void onImageClick();

        void onOptionsClick();
        void onDeleteClick();
        void onSendClick();
        void onDuplicateClick();
        void onLabelClick();
        void onColorSelected(NoteColor color);
        void onPlaceSelected(Place place);
    }
}

File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/note/presenter/NoteContract.kt
package com.github.abhijitpparate.keeper.screen.note.presenter


import android.support.annotation.ColorRes
import android.support.annotation.StringRes

import com.github.abhijitpparate.checklistview.CheckListItem
import com.github.abhijitpparate.keeper.data.database.Note
import com.google.android.gms.location.places.Place

interface NoteContract {

    enum class NoteColor {
        DEFAULT, WHITE, PURPLE, RED, GREEN, YELLOW, BLUE, ORANGE
    }

    interface View {

        fun getNoteUuid(): String?
        fun getNoteTitle(): String?
        fun getNoteBody(): String?
        fun getNoteCheckList(): String?
        fun getNoteColor(): String?
        fun getNotePlace(): Note.Place?

        fun setNote(note: Note)
        fun setNoteTitle(title: String?)
        fun setNoteBody(body: String?)
        fun setNoteChecklist(checklist: List<CheckListItem>)
        fun setNoteColor(@ColorRes colorInt: Int, colorString: String)
        fun setNoteLocation(place: Note.Place)

        fun showOptionsPanel()
        fun hideOptionsPanel()

        fun showLocationPicker()

        fun showHomeScreen()

        fun loadNoteIfAvailable()

        fun setPresenter(presenter: NoteContract.Presenter)
        fun showProgressbar(bool: Boolean)
        fun makeToast(@StringRes strId: Int)
        fun makeToast(message: String)

        fun switchToChecklist()
        fun switchToText()
    }

    interface Presenter {
        fun onSaveClick()

        fun loadNote(noteId: String)

        fun subscribe()
        fun unsubscribe()

        fun onChecklistClick(isChecked: Boolean)
        fun onDrawingClick()
        fun onLocationClick()
        fun onAudioClick()
        fun onVideoClick()
        fun onImageClick()

        fun onOptionsClick()
        fun onDeleteClick()
        fun onSendClick()
        fun onDuplicateClick()
        fun onLabelClick()
        fun onColorSelected(color: NoteColor)
        fun onLocationSelected(location: Place)
    }
}
